src/app/workflow/launch-third-party/launch-third-party.component.ts

```diff
@@ -319,6 +292,19 @@ export class LaunchThirdPartyComponent extends Base implements OnChanges, OnInit
       this.workflowRepositoryAsQueryValue = this.encode(this.workflow.repository);
       this.selectedVersionNameAsQueryValue = this.encode(this.selectedVersion.name);
       this.selectedVersionWorkflowPathAsQueryValue = this.encode(this.prependIfNotPrefix('/', this.selectedVersion.workflow_path));
+      const descriptorType = this.workflow.descriptorType;
+      const descriptorLanguageEnum =
+        this.descriptorLanguageService.workflowDescriptorTypeEnumToExtendedDescriptorLanguageBean(descriptorType).descriptorLanguageEnum;
+      this.primaryDescriptorSubscription = this.workflowsService
+        .primaryDescriptor1(this.workflow.id, descriptorType, this.selectedVersion.name)
+        .subscribe((sourceFile) => {
+          this.descriptorsService.updatePrimaryDescriptor(sourceFile);
+        });
+      this.secondaryDescriptorsSubscription = this.workflowsService
```

Comment:
Before your PR, this request was being done after the request to fetch the primary descriptor. In your PR it's being done in parallel.

Just checking if we're OK switching that. It is broken now, and this approach should be faster, so I suppose it's OK.

Reply:
Yeah, the tradeoff here is we might do an extra request or two, but they are initiated earlier and in parallel.  I think the new way is ok, but we'll need to make sure it doesn't feel any laggier after we've deployed to staging/production.

Reply:
> Yeah, the tradeoff here is we might do an extra request or two, but they are initiated earlier and in parallel. I think the new way is ok, but we'll need to make sure it doesn't feel any laggier after we've deployed to staging/production.

My edge-case concern was more that there was some weird UI interaction, e.g., the display of the secondary descriptors somehow depended on the primary descriptor being displayed first. Just wanted to make triple-sure... But I think it's OK.